### PR TITLE
Add support for container embedding

### PR DESCRIPTION
**Pull request by gicmo**
This add support for embedding containers.

TODO in this PR:
 - [x] RHEL 8
 - ~[ ] Fedora~
 - [x] Testing
 - [x] Commit messages

### CODE REVIEW

**Review by achilleas-k — COMMENTED** [review]
Good stuff.  Some typos below.

The resolver queue is a bit unusual, but simple enough that it doesn't matter really.

Is there a reason the result queue (channel) has a size of 2 specifically?

**Comment by teg** [comment]
For the uninitiated, could you give a high level overview of the difference between manifest and digest.

Am I right in thinking digest is a content hash, and that manifest + arch uniquely identify a digest? Or is it more like a git tag so the digest can change without the manifest changing?

Is referencing things by manifest always preferable to digest in common tooling, if that's even a thing? If that's not a thing, why not?

**Comment by achilleas-k** [comment]
Package naming errors otherwise looks great!

**Review by ondrejbudai — COMMENTED** [review]
Very nice!

File: internal/container/container_test.go [hunk 1]
```diff
@@ -0,0 +1,382 @@
...
+	}
+
+	if cmd == "manifests" {
+		repo.ServeManifest(ref, w, req)
+	} else if cmd == "blobs" {
+		repo.ServeBlob(ref, w, req)
+	} else {
+		http.NotFound(w, req)
+	}
+}
```
[the remark this answers is not in the document]
Comment (gicmo) [reply internal/container/container_test.go:288]: Yeah, decided against that to have more control over the path parsing as seen in the lines above.

**Comment by gicmo** [comment]
Decided to do Fedora in a follow up PR.

**Review by achilleas-k — APPROVED** [review]
All looks great!

**Comment by ondrejbudai** [comment]
`aws.sh` on Fedora is currently broken, fix already landed on `main` in https://github.com/osbuild/osbuild-composer/pull/2782. This is fine to be merged.

**Pull request merged**